src/event.rs

```diff
@@ -2,400 +2,288 @@ use app::ScrollDirection;
                 }
             }
-            KeyCode::Char('?') => {
-                app.previous_mode = app.mode;
-                app.mode = app::Mode::Help;
-                let _ = request_redraw.try_send(());
+        }
+        KeyCode::Left => {
+            app.stocks[app.current_tab]
+                .options
+                .as_mut()
+                .unwrap()
+                .selection_mode_left();
+        }
+        KeyCode::Right => {
+            if app.stocks[app.current_tab]
+                .options
+                .as_mut()
+                .unwrap()
+                .data()
+                .is_some()
+            {
+                app.stocks[app.current_tab]
+                    .options
+                    .as_mut()
+                    .unwrap()
+                    .selection_mode_right();
             }
-            _ => {}
         }
-    } else if key_event.modifiers == KeyModifiers::CONTROL {
-        if let KeyCode::Char('c') = key_event.code {
+        _ => {}
+    }
+}
+
+pub fn handle_key_bindings(
+    mode: Mode,
+    key_event: KeyEvent,
+    mut app: &mut app::App,
+    request_redraw: &Sender<()>,
+) {
+    match (mode, key_event.modifiers, key_event.code) {
+        (_, KeyModifiers::CONTROL, KeyCode::Char('c')) => {
+            cleanup_terminal();
+            std::process::exit(0);
+        }
+        (Mode::AddStock, modifiers, keycode)
+            if modifiers.is_empty() || modifiers == KeyModifiers::SHIFT =>
+        {
+            handle_keys_add_stock(keycode, app)
+        }
+        (Mode::Help, modifiers, keycode)
+            if modifiers.is_empty()
+                && (matches!(
+                    keycode,
+                    KeyCode::Esc | KeyCode::Char('?') | KeyCode::Char('q')
+                )) =>
+        {
+            app.mode = app.previous_mode;
+        }
+        (mode, _, KeyCode::Char('q')) if mode != Mode::DisplayOptions => {
             cleanup_terminal();
             std::process::exit(0);
         }
-    } else if key_event.modifiers == KeyModifiers::SHIFT {
-        if let KeyCode::Char('?') = key_event.code {
+        (.., KeyCode::Char('?')) => {
             app.previous_mode = app.mode;
             app.mode = app::Mode::Help;
-            let _ = request_redraw.try_send(());
         }
+        (.., KeyCode::Char('c')) => {
+            let mut chart_type = CHART_TYPE.write().unwrap();
+            *chart_type = chart_type.toggle();
+        }
+        (.., KeyCode::Char('v')) => {
+            let mut show_volumes = SHOW_VOLUMES.write().unwrap();
+            *show_volumes = !*show_volumes;
+        }
+        (.., KeyCode::Char('p')) => {
+            let mut guard = ENABLE_PRE_POST.write().unwrap();
+            *guard = !*guard;
+        }
+        (Mode::DisplaySummary, modifiers, keycode) if modifiers.is_empty() => {
+            handle_keys_display_summary(keycode, app)
+        }
+        (.., KeyCode::Char('x')) => {
```

Comment:
This doesn't affect the view in Summary mode so we shouldn't have it captured there

Reply:
It's matched after the `DisplaySummary` for this reason. So, on the summary view, the match condition directly above this one is branched into. So, it never gets here and never toggles `x` unless there are no modifiers.

Reply:
It's matched in top-down order as enumerated in the table https://github.com/tarkah/tickrs/pull/95#pullrequest-579985003-permalink:~:text=keymap%20(top%2Ddown%20in%20match%20order)